packages/ckeditor5-html-embed/src/htmlembedediting.js

```diff
@@ -165,6 +166,14 @@ export default class HtmlEmbedEditing extends Plugin {
 						// If the value didn't change, we just cancel. If it changed,
 						// it's enough to update the model – the entire widget will be reconverted.
 						if ( newValue !== state.getRawHtmlValue() ) {
+							const selectedElement = model.document.selection.getSelectedElement();
+							const modelContainer = editor.editing.mapper.toModelElement( viewContainer );
+
+							// The HTML embed widget must be selected to be able to save new value.
+							if ( selectedElement !== modelContainer ) {
+								model.change( writer => writer.setSelection( modelContainer, 'on' ) );
+							}
```

Comment:
I'm not sure if this is the right place for this. This widget focus behaves quite inconsistent (because of the `data-cke-ignore-events` feature, but it's needed in editing mode).

Have a look at this screencast:
![2020-11-19 16 38 43](https://user-images.githubusercontent.com/1232187/99691007-094f2500-2a89-11eb-8614-0700ab701688.gif)
While not in editing mode - click on the widget is handled by the [`mousedown` handler in the widget plugin](https://github.com/ckeditor/ckeditor5/blob/master/packages/ckeditor5-widget/src/widget.js#L187), but in the editing mode those events are blocked by `data-cke-ignore-events`. Maybe this should be handled in a similar way in this feature - the selection might be moved onto the widget before a button click was handled or a textarea focused?

Reply:
That would solve other issues as well. I'm thinking about handling this in other PR (#8421).

Similar case: #8328.

Reply:
I've changed my solution basing on your comment (I hope I understood you correctly 😅), and I've added a `click` event listener on the `capture` phase to the wrapper to be able to move the selection onto the whole widget in the editing mode.